crates/oxc_parser/src/ts/statement.rs

```diff
@@ -422,19 +422,11 @@ impl<'a> Parser<'a> {
         ))
     }
 
-    pub(crate) fn parse_ts_this_parameter(&mut self) -> Result<FormalParameter<'a>> {
+    pub(crate) fn parse_ts_this_parameter(&mut self) -> Result<TSThisParameter<'a>> {
         let span = self.start_span();
-        let (ident_span, name) = self.parse_identifier_kind(Kind::This);
+        let _ = self.parse_identifier_kind(Kind::This);
```

Comment:
Does it make sense to keep the `this_ident` in AST?

Reply:
yeah it's good to add the `this` identifier to the AST.